src/component/tooltip/TooltipView.ts

```diff
@@ -505,18 +512,20 @@ class TooltipView extends ComponentView {
                     const series = ecModel.getSeriesByIndex(idxItem.seriesIndex);
                     const dataIndex = idxItem.dataIndexInside;
                     const cbParams = series.getDataParams(dataIndex) as TooltipCallbackDataParams;
+                    cbParams.name = encode ? encodeHTML(cbParams.name) : cbParams.name;
                     cbParams.axisDim = axisItem.axisDim;
                     cbParams.axisIndex = axisItem.axisIndex;
                     cbParams.axisType = axisItem.axisType;
                     cbParams.axisId = axisItem.axisId;
-                    cbParams.axisValue = axisHelper.getAxisRawValue(
+                    const cbAxisValue = axisHelper.getAxisRawValue(
                         axisModel.axis, { value: axisValue as number }
-                    );
+                    ) as string;
+                    cbParams.axisValue = encode ? encodeHTML(cbAxisValue) : cbAxisValue;
```

Comment:
I think we should not `encodeHTML` here.
`encodeHTML` only make sense in HTML rendering.
The encoded string should not be exposed to users in callback.
That is, when we provide the "non-encoded" string to users, 
users can encode it themselves if needed.
But if we provide an "encoded" string, user can not do anything others to them.
Especially when the `renderMode` is `richText` rather than `html`,
there should be no encode.

I think #14429 is probably not an issue.
See the [comment](https://github.com/apache/echarts/issues/14429#issuecomment-794847914) I've added just now.